modules/kibana/build.gradle

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+esplugin {
+  description 'Plugin exposing APIs for Kibana system indices'
+  classname 'org.elasticsearch.kibana.KibanaPlugin'
+}
+
+dependencies {
+  compile project(path: ':modules:reindex', configuration: 'runtime')
```

Comment:
This caught my eye and after a quick search I see we do this all over the place. @rjernst any idea why were are explicitly declaring a dependency on the publishing projects `runtime` configuration? This is generally redundant as the default configuration for Java projects typically is this already.

Reply:
Well actually, I think what the behavior of this is, is that we will bring in all the runtime _dependencies_ of that project, but not the JAR for that project itself. Is that handled differently? Are we expecting the module jars to be on the runtime classpath already so no need?

Reply:
I don't recall why we have this in some places. I do now notice we explicitly declare the default configuration to extend runtime (in PluginBuildPlugin):

```
project.configurations.getByName('default').extendsFrom(project.configurations.getByName('runtime'))
```

I vaguely recall having issues with getting the jars instead of the zip. In the case here, we are trying to reuse the request builders, but unfortunately we don't package or ship these, so we must pull in a copy of the entire module.

Reply:
> I vaguely recall having issues with getting the jars instead of the zip. 

Ah, this sounds like it makes sense. Thanks.